ext/afform/core/afform.php

```diff
@@ -127,6 +127,33 @@ function afform_civicrm_upgrade($op, CRM_Queue_Queue $queue = NULL) {
  */
 function afform_civicrm_managed(&$entities) {
   _afform_civix_civicrm_managed($entities);
+
+  $afforms = \Civi\Api4\Afform::get(FALSE)
+    ->setSelect(['title', 'name', 'module_name', 'directive_name'])
+    ->addWhere('is_dashlet', '=', TRUE)
+    ->execute();
+  $afforms = $afforms->getArrayCopy();
+
+  foreach ($afforms as $afform) {
+    $entities[] = [
+      'module' => E::LONG_NAME,
+      'name' => 'dashlet_' . $afform['name'],
+      'entity' => 'Dashboard',
+      'update' => 'always',
+      // ideal cleanup policy might be to (a) deactivate if used and (b) remove if unused
+      'cleanup' => 'unused',
+      'params' => [
+        'version' => 3,
+        // Q: Should we loop through all domains?
```

Comment:
How does `hook_civicrm_managed` interact with domains? Does it run once per domain, or... ?

Reply:
I'm pretty sure `hook_civicrm_managed` is completely ignorant of domains. It probably should be -- because it aims to work with all entities -- some entities (like `Dashboard`) are domain-oriented and others (like `Afform`) aren't. (Also, `ManagedEntities.php` doesn't have the word "domain" anywhere in it.)

So if we wanted each form's dashlet to appear in all domains, then we'd need to map each one-by-one.

I guess another option might be to change the dashboard subsystem to allow `domain_id=null` as a wildcard?